compiler/src/main/kotlin/se/ansman/kotshi/AdaptersProcessingStep.kt

```diff
@@ -250,7 +250,8 @@ class AdaptersProcessingStep(
                             addStatement("$adapterName.toJson(writer, $getter)")
                         } else {
                             fun MethodSpec.Builder.writePrimitive(getter: String) =
-                                    when (property.type.unbox()) {
+                                    when (if (property.type.isBoxedPrimitive) property.type.unbox() else property.type) {
```

Comment:
Can we pull this conditional into an extension property since it's used twice? Maybe called `readerType` or `jsonReaderType`?

Reply:
Good point, done